### ISSUE TITLE
Lost UI Shared Context

### ISSUE THREAD

**Issue by tleunen** [puppeteer/puppeteer#1925]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 1.0.0
* Platform / OS version: puppeteer docker
* URLs (if applicable):
* Node.js version: 8

**What steps will reproduce the problem?**

It's really hard to know what is causing the issue.
I'm running the test in docker with this command:
```
docker run -i --rm --cap-add=SYS_ADMIN --memory=8g -v "$HOSTDIR:$CLIENTDIR" web-puppeteer /bin/bash $CLIENTDIR/script.sh
```
With script.sh being just `npm i` and `npm t`, which runs Karma with puppeteer as a browser. Here's the custom browser config:
```
ChromeHeadlessCustom: {
        base: 'ChromeHeadless',
        flags: ['--no-sandbox', '--disable-setuid-sandbox', '--disable-gpu', '--disable-dev-shm-usage']
      }
```

I thought the issue was because of a memory issue, so I increased it, but it doesn't seem to resolve it

**What is the expected result?**
Running a karma test suite with puppeteer shouldn't cause any issues

**What happens instead?**
ChromeHeadless crashes with this error:
```
build	30-Jan-2018 17:05:26	[31m30 01 2018 17:05:26.258:ERROR [launcher]: [39mChromeHeadless crashed.
build	30-Jan-2018 17:05:26		[0130/170455.872164:ERROR:gpu_process_transport_factory.cc(1043)] Lost UI shared context.
```

**Comment by melroy89** [comment]
Yea uhh same problem.. +1

*UPDATE:* I'm currently using a correct Docker image instead, which has everything correctly setup. Using the **headless** alpine tag: `weboaks/node-karma-protractor-chrome:alpine-node9`. You can also use Debian based images, use: `headless` tag.

Carefully follow the steps for configuration:
https://hub.docker.com/r/weboaks/node-karma-protractor-chrome/

**Comment by ebidel** [comment]
What's the base image of your docker? Can you provide a zip or repo that has all the pieces to repro.

**Comment by melroy89** [comment]
I'm/was using the official node docker image (latest):
https://hub.docker.com/r/_/node/

Currently trying `weboaks/node-karma-protractor-chrome:alpine-node9` from the hub as well. This solves the issue.

**Comment by sylvaindumont** [comment]
These errors seem related to chrome 64, with `weboaks/node-karma-protractor-chrome` chrome 63 have no errors.
It happens on alpine and debian.

`weboaks/node-karma-protractor-chrome:alpine-6a3d354178f4a2c23df0ea9d5ee0f52e940e0672` and `weboaks/node-karma-protractor-chrome:alpine` are both build with this [dockerfile](https://github.com/sylvaindumont/docker-node-karma-protractor-chrome/blob/6a3d354178f4a2c23df0ea9d5ee0f52e940e0672/Dockerfile). 

On `alpine-6a3d354178f4a2c23df0ea9d5ee0f52e940e0672` built 2 month ago with chromium 63 we can run `chromium --disable-gpu --headless --no-sandbox` with no errors. 
On `alpine` with chromium 64, the same command give these errors
```
[0314/215407.941298:ERROR:gpu_process_transport_factory.cc(1009)] Lost UI shared context.
[0314/215408.389043:ERROR:instance.cc(49)] Unable to locate service manifest for metrics
[0314/215408.396213:ERROR:service_manager.cc(890)] Failed to resolve service name: metrics
[0314/215408.454039:ERROR:gl_implementation.cc(292)] Failed to load /usr/lib/chromium/swiftshader/libGLESv2.so: Error loading shared library /usr/lib/chromium/swiftshader/libGLESv2.so: No such file or directory
[0314/215408.456937:ERROR:viz_main_impl.cc(196)] Exiting GPU process due to errors during initialization
```

On `debian-9bf3ee4fb2049652de8a4502e1cb210bf6c42079` and `debian` are both build with this [dockerfile](https://github.com/sylvaindumont/docker-node-karma-protractor-chrome/blob/9bf3ee4fb2049652de8a4502e1cb210bf6c42079/Dockerfile), the first with chromium 63 have no errors and the second with chromium 64 have these errors
```
[0314/215541.875522:ERROR:gpu_process_transport_factory.cc(1009)] Lost UI shared context.
[0314/215541.875531:ERROR:instance.cc(49)] Unable to locate service manifest for metrics
[0314/215541.875737:ERROR:service_manager.cc(890)] Failed to resolve service name: metrics
```

**Comment by remborg** [comment]
To complete what sylvaindumont said just above, I'm having the same issue with chrome 65 running locally on a win7-64

**Comment by damienwebdev** [comment]
Just to add to this, I'm seeing the same thing on a fresh Ubuntu 16.04 vm.

## Steps to reproduce

### Tell us about your environment:
Puppeteer version: 1.2.0 or 1.0.0 (I tried both)
Platform / OS version: Ubuntu 16.04
Linux Kernel: 4.13.0-1011-azure
Node.js version: v8.10.0
Chrome rev from installed package.json:
```
"puppeteer": {
    "chromium_revision": "543305"
},
```

`node_modules/puppeteer/.local-chromium/linux-543305/chrome-linux/chrome --user-data-dir=/tmp/karma-31644641 --no-default-browser-check --no-first-run --disable-default-apps --disable-popup-blocking --disable-translate --disable-background-timer-throttling --disable-renderer-backgrounding --disable-device-discovery-notifications --headless --disable-gpu`

`[0322/022058.750959:ERROR:gpu_process_transport_factory.cc(1008)] Lost UI shared context.`

**Comment by hashar** [comment]
> Failed to load /usr/lib/chromium/swiftshader/libGLESv2.so: Error loading shared library /usr/lib/chromium/swiftshader/libGLESv2.so: No such file or directory

That is most probably https://bugs.debian.org/864606 . Should be fixed with Chromium v66

**Comment by Storyyeller** [comment]
I'm still getting this error message with Chrome 68.

**Comment by stopsopa** [comment]
I'm using this Dockerfile: https://raw.githubusercontent.com/stopsopa/docker-puppeteer-html-scraper/master/Dockerfile

and the same problem

**Comment by JonWallsten** [comment]
Getting this with CentOS 7.5 and latest Puppeteer (1.5.0), that should be Chrome 69 if I'm not mistaken.

**Comment by pgebheim** [comment]
I am currently getting this with a number of configurations and chrome versions under docker.

The command line flags are what we came to that would actually not throw random GPU errors while trying to load libglesv2 on the systems. It turns out --no-gpu doesn't actually disabled the GPU!

So @sylvaindumont 's error above on alpine (related to not being able to load libglesv2) is solved by passing in `--disable-software-rasterizer`.

Notes:
- All of these are configured using jest-chrome. https://github.com/AugurProject/augur/blob/docker-integration-test/jest-puppeteer.config.js
- All configurations have been dockerized
- Examine the docker files to show that we've updated all necessary dependencies based on the numerous other docker images of people attempting to get this working.
- Running is done via docker-compose (also found in the linked repository), and SYS_ADMIN privledge escalation has been applied.

## Alpine Linux
Puppeteer Version: 0.13.0
Chrome Revisions: 64
Node Version: 8 or 9
Configuration:
- Problem occurs using the puppeteer-downloaded chrome, as well as the system chrome.
- Running chrome with: `--no-gpu --disable-software-rasterizer --headless --mute-audio --hide-scrollbars --remote-debugging-port=9222`
Links:
 - Dockerfile: https://github.com/AugurProject/augur/blob/docker-integration-test/Dockerfile-integration

## Base: Debian 8 (Jessie) - Slim
Docker Image: nodejs/node:8-slim
Puppeteer Version(s): 1.4.0, 1.5.0
Chrome Revisions: 68 (puppeteer 1.4.0), 69 (1.5.0) -- using system chrome and downloaded chrome
Node version: 9
Configuration:
- Problem occurs using the puppeteer-downloaded chrome, as well as the system chrome.
- Running chrome with: `--no-gpu --disable-software-rasterizer --headless --mute-audio --hide-scrollbars --remote-debugging-port=9222`
Links:
 - Dockerfile: https://github.com/AugurProject/augur/blob/docker-integration-test/Dockerfile-integration-slim

## Base: Ubuntu 16.04 (Xenial)
Docker Image: using the selenium base image `selenium/node-chrome`
Puppeteer Version: 1.5.0
Chrome Revisions: 69 installed on system, 68 installed by p
Node version: 8
Configuration:
- Problem occurs using the puppeteer-downloaded chrome, as well as the system chrome.
- Running chrome with: `--no-gpu --disable-software-rasterizer --headless --mute-audio --hide-scrollbars --remote-debugging-port=9222`
- Links:
  - Dockerfile: https://github.com/AugurProject/augur/blob/use-selenium-chrome-container/Dockerfile-integration

**Comment by kenorbi** [comment]
same problem here on ubuntu 16.04, any slutions?

**Comment by matAtWork** [comment]
+1 on Ubuntu 14.04

**Comment by pYr0x** [comment]
same on
`docker run -it --rm -p=0.0.0.0:9222:9222 --name=chrome-headless alpeware/chrome-headless-trunk`
with 
`docker exec -it chrome-headless /usr/bin/google-chrome-unstable --disable-gpu --headless --no-sandbox --disable-dev-shm-usage --disable-software-rasterizer http://google.de`

**Comment by NCarson** [comment]
When I run `chrome --headless` from the command line it prints the error but it still works and exits 0. 

This guy says it will only effect WebGL and  to just ignore it for now:
https://stackoverflow.com/questions/50143413/errorgpu-process-transport-factory-cc1007-lost-ui-shared-context-while-ini

**Comment by peterver** [comment]
same problem on Arch Linux with Kernel 4.17.8-1 here.

Output : 
```
gpu_process_transport_factory.cc(1007) Lost UI shared context
```

Running through karma with the following config:
```
process.env.CHROME_BIN = require('puppeteer').executablePath();

module.exports = function(config) {
    config.set({
        basePath: './',
        frameworks: ['jasmine'],
        files: [
            {pattern: 'test/test_index.js'},
        ],
        exclude: [],
        preprocessors: {
            'test/test_index.js': ['webpack'],
        },
        webpack: {
            mode: 'development',
        },
        webpackMiddleware: {
            noInfo: true,
            stats: {
                chunks: false,
            },
        },
        reporters: ['progress'],
        port: 9876,
        colors: true,
        logLevel: config.LOG_INFO,
        autoWatch: true,
        browsers: ['ChromeHeadless'],
        singleRun: false,
        customLaunchers: {
            ChromeHeadless: {
                base: 'Chrome',
                flags: ['--disable-gpu', '--headless', '--no-sandbox', '--disable-dev-shm-usage', '--disable-software-rasterizer'],
            }
        }
    });
};
```

**Comment by cztomczak** [comment]
From what I've read on the web and seen in my app the `Lost UI Shared Context` error message can be safely ignored and is probably not the cause of the crash.

**Comment by NCarson** [comment]
depends on what you want. Maybe the Googly does not want you to have access to the tools it has.

**Comment by jlandure** [comment]
I agree with @NCarson @aslushnikov  
> these are fine :) They come from chromium internals and are probably used by the chromium developers.

Have a look https://github.com/Zenika/alpine-chrome/issues/6

**Comment by xbmono** [comment]
We have the same issue in Jenkins. It works fine in Bitbucket Pipeline but the same thing (same image) doesn't work in Jenkins installed in AWS. 
This is the error message

```
[emberTests3] not ok 1 Chrome - error
[emberTests3]     ---
[emberTests3]         message: >
[emberTests3]             Error: Browser exited unexpectedly
[emberTests3]             Stderr: 
[emberTests3]              [1011/065617.659548:ERROR:gpu_process_transport_factory.cc(1016)] Lost UI shared context.
[emberTests3]             [1011/065617.663400:ERROR:zygote_host_impl_linux.cc(270)] Failed to adjust OOM score of renderer with pid 1001: Permission denied (13)
[emberTests3]             Fontconfig warning: "/etc/fonts/fonts.conf", line 146: blank doesn't take any effect anymore. please remove it from your fonts.conf
[emberTests3]             
[emberTests3]             
[emberTests3]         Log: |
[emberTests3]             { type: 'error', text: 'Error: Browser exited unexpectedly' }
[emberTests3]             { type: 'error',
[emberTests3]               text: '[1011/065617.659548:ERROR:gpu_process_transport_factory.cc(1016)] Lost UI shared context.\n[1011/065617.663400:ERROR:zygote_host_impl_linux.cc(270)] Failed to adjust OOM score of renderer with pid 1001: Permission denied (13)\nFontconfig warning: "/etc/fonts/fonts.conf", line 146: blank doesn\'t take any effect anymore. please remove it from your fonts.conf\n' }
```

We are using Ember Exam to run the tests

This is my testem.js config:

```
module.exports = {
    "test_page": "tests/index.html?hidepassed",
    "disable_watching": true,
    "parallel": 6,
    "launch_in_ci": [
        "Chrome"
    ],
    "launch_in_dev": [
        "Chrome"
    ],
    "browser_start_timeout": 80,
    "browser_args": {
        "Chrome": [
                '--no-sandbox',
                '--headless',
                '--window-size=1440,900'
            ]
    }
};
```

**Comment by scholtzm** [comment]
This has started happening constantly on my Macbook Pro after updating to v1.9.0 via npm.

**Tell us about your environment:**

- Puppeteer version: 1.9.0; installed via npm
- Platform / OS version: macOS 10.13.6 (High Sierra)
- Node.js version: 8.12.0
- Launcher: karma

I have also tested this on another Macbook Pro (macOS Mojave) and it's having the same issue.

**Comment by dampcake** [comment]
Same environment and same problem that @scholtzm is having. Consistently failing after updating to 1.9.0.

**Comment by ydnar** [comment]
Same. I can’t run any tests with 1.9.0. Puppeteer consistently fails with this error.

**Comment by chris-aeviator** [comment]
Can confirm that applying "--disable-dev-shm-usage" && "--disable-software-rasterizer" solves the problem for my Dockerrized Alpine CI environment

**Comment by aslushnikov** [comment]
**TL;DR: You'll never see this error any more starting from Chrome 72. ([upstream patch](https://chromium-review.googlesource.com/c/chromium/src/+/1333832/2))**

Closing this thread since it became a dumping ground for all kind of unrelated bug reports. 

### Explanation 

The "Lost UI Shared Context" is not an error - that's just a logging output that [comes from chrome internals](https://cs.chromium.org/chromium/src/content/browser/compositor/gpu_process_transport_factory.cc?type=cs&q=%22lost+ui+shared+context%22&sq=package:chromium&g=0&l=963). It's a valid codepath inside Chromium, and it's unfortunate that this gets logged as error - and it's been fixed upstream.

### Q: But my browser still crashes!

If browser crashes - there is another reason for it to do so, and usually it will be printed below the "Lost UI Shared Context". 

@xbmono in your case, the issue is listed below:

```
Failed to adjust OOM score of renderer with pid 1001: Permission denied (13)
```

I'm not very familiar with this corner of Chromium codebase, but it looks like Chromium is either tight on Memory or doesn't have sufficient permissions. Please file a separate bug to continue discussion.

@scholtzm @dampcake @ydnar your case is not clear to me - the "Lost ui shared context" is not an actual error, so the full chromium output is needed to see what's going where. Please, file separate bugs with full error log from your crashed browser instances.

**Comment by perry-mitchell** [comment]
For those still getting this issue (I'm seeing it on Chrome 70), adding `--remote-debugging-port=9222` fixed it for me. Trying to go headless without this resulted in the error immediately.

**Comment by remmel** [comment]
I also got that message in _Google Chrome 71.0.3578.98_ on _Ubuntu 18.10_  with whatever command, eg:
```
chrome --headless --disable-gpu --dump-dom https://www.chromestatus.com/
[0102/135155.575491:ERROR:gpu_process_transport_factory.cc(967)] Lost UI shared context.
[0102/135156.382642:INFO:headless_shell.cc(546)] Written to file screenshot.png.
```

Adding remote-debugging-port as @perry-mitchell suggest doesn't change anything

**Comment by Megagyger** [comment]
@perry-mitchell solution solved for me
RHEL 7.4 Maipo
google-chrome-stable-71.0.3578.98-1.x86_64

Karma Flags for the launcher
 --no-sandbox --headless --disable-gpu --disable-software-rasterizer --disable-dev-shm-usage --remote-debugging-port=9222

**Comment by aryascripts** [comment]
@perry-mitchell solution worked!
Current flags for base 'Chrome' using an ubuntu docker image (private)

```
        flags: [
          '--headless',
          '--disable-gpu',
          // Without a remote debugging port, Google Chrome exits immediately.
          '--remote-debugging-port=9222',
          '--no-sandbox',
          '--disable-software-rasterizer',
          '--mute-audio',
          '--hide-scrollbars',
          '--disable-dev-shm-usage'
        ]
```

**Comment by luckman212** [comment]
> **TL;DR: You'll never see this error any more starting from Chrome 72.**

Confirmed that Chrome 72.0.3626.81 released today fixes this (cosmetic) issue. Thank you

**Comment by shoesCodeFor** [comment]
Thanks @perry-mitchell.  This solved my issue as well.  Docker Image ruby:2.6.1-alpine3.9

`chromium-browser --headless --disable-gpu --disable-software-rasterizer --disable-dev-shm-usage --no-sandbox --remote-debugging-port=9222
`

**Comment by nojvek** [comment]
`/usr/bin/google-chrome-stable --headless --disable-gpu --window-size=1280,800 --disable-dev-shm-usage --disable-software-rasterizer --no-sandbox`

Doesn't work for me.

```
Fontconfig warning: "/etc/fonts/fonts.conf", line 86: unknown element "blank"
[0213/062220.288103:ERROR:gpu_process_transport_factory.cc(967)] Lost UI shared context.
```

**Comment by JonWallsten** [comment]
> `/usr/bin/google-chrome-stable --headless --disable-gpu --window-size=1280,800 --disable-dev-shm-usage --disable-software-rasterizer --no-sandbox`
> 
> Doesn't work for me.
> 
> ```
> Fontconfig warning: "/etc/fonts/fonts.conf", line 86: unknown element "blank"
> [0213/062220.288103:ERROR:gpu_process_transport_factory.cc(967)] Lost UI shared context.
> ```

@nojvek For the first warning you need to go into the fonts file and remove the <blank>-tag.
The second line by itself is not an issue. I still get it in my logs even though everything works.

**Comment by tiagopmc** [comment]
I`m still having this problem running on aws lambda. I`m using this flags:
 '--headless',
          '--disable-gpu',
          '--disable-software-rasterizer',
          '--disable-dev-shm-usage',
          '--no-sandbox',
          '--remote-debugging-port=9222'

**Comment by PJUllrich** [comment]
`chromium-browser --headless --disable-gpu --disable-software-rasterizer --no-sandbox` did the trick for me.
